server/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -789,6 +789,7 @@ public Object readGenericValue() throws IOException {
             case 27 -> readOffsetTime();
             case 28 -> readDuration();
             case 29 -> readPeriod();
+            case 30 -> readNamedWriteable(NamedWriteable.class);
```

Comment:
This makes me a little uncomfortable, we should use a more precise class than just `NamedWriteable`. But I see why it's tricky here when we really don't know the context in which we're reading.

Could we introduce a marker type called `GenericNamedWriteable` and use that instead of `NamedWriteable`? That would help to protect against just accepting absolutely any `NamedWriteable` here.

Also I think this counts as a new transport version, we need to add a new version constant and throw an exception if the remote node version is too old rather than sending it something it won't understand.

Reply:
Great advice. I'm creating GenericNamedWriteable to extend VersionedNamedWriteable, which also provides the version checking.

Reply:
I recommend not using `VersionNamedWriteable`, it is kind of a bad idea (it's designed to just silently drop stuff when sending to older versions, and that's almost certainly a mistake).

Reply:
Actually wait, I see, it could be useful here. But we do need to throw something (and ideally `assert false` too) if the sender is trying to send a value that the receiver won't understand.